numpy/typing/__init__.py

```diff
@@ -68,13 +68,26 @@
 
 .. code-block:: python
 
-    x = np.array([1, 2])
-    x.dtype = np.bool_
+    >>> x = np.array([1, 2])
+    >>> x.dtype = np.bool_
 
 This sort of mutation is not allowed by the types. Users who want to
 write statically typed code should insted use the `numpy.ndarray.view`
 method to create a view of the array with a different dtype.
 
+dtype
+~~~~~
+
+The ``_DTypeLike`` type tries to avoid creation of dtype objects using
```

Comment:
Note that ``_DTypeLike`` is not strictly speaking a type; it's a special form (_i.e._ ``typing.Union``).
Maybe "annotation" might be a better term here?

Reply:
hmm, not sure.  i consider even Unions to be types and I made it similar to ArrayLike note above. Though, annotation is less likely to confuse maybe ? Do you suggest we change both _Dtypelike and ArrayLike to annotation ?

Reply:
> Do you suggest we change both _Dtypelike and ArrayLike to annotation ?

Something like that, yes.
Whether or not it would change into "annotation" or something else, 
I feel it's important enough to clarify that we're not dealing with `type` objects here.

Reply:
i changed type to static type to make the distinction more prominent. Please let me know if you have other suggestions to make it more clear.

Reply:
I'm not sure that I follow this discussion. How is union not a type? From e.g. the docs:

https://docs.python.org/3/library/typing.html#typing.Union

> Union type; Union[X, Y] means either X or Y.

I don't think adding "static" makes it more clear; I think it makes it less clear.

Reply:
I reverted the static type change. I understand that this can cause a bit of confusion since these arent type objects and I see that PEP-484 and PEP-483 use "static type" to differentiate with runtime classes. But yes the typing module doc doesnt define "static type" explicitly and it can cause some confusion. Since these appear in numpy.typing docs where we are talking about PEP-484 style type annotations and we have already mentioned they are type aliases before, maybe "type" is just fine.